### PR TITLE
Frozen esdt tokens

### PR DESCRIPTION
**Pull request by miiu96**
- Added an extra field `frozen` in the `tokens` index ( is true if an NFT is frozen)
- Added an extra field `paused` in the `tokens` index ( is true if an ESDT token is paused)
- Extended the `accountsesdt` index

### CODE REVIEW

File: process/elasticproc/logsevents/nftsPropertiesProcessor.go [hunk 1]
```diff
@@ -70,6 +78,26 @@ func (npp *nftsPropertiesProc) processEvent(args *argsProcessEvent) argOutputPro
...
+		updatePropNFT: updateNFT,
+	}
+}
+
+func (npp *nftsPropertiesProc) processPauseAndUnPauseEvent(eventIdentifier string, token string) argOutputProcessEvent {
+	var updateNFT *data.NFTDataUpdate
+
+	switch eventIdentifier {
+	case core.BuiltInFunctionESDTPause:
+		updateNFT = &data.NFTDataUpdate{Identifier: token, Pause: true}
```
Comment (iulianpascalau) [inline process/elasticproc/logsevents/nftsPropertiesProcessor.go:98]: could have been multiline. Also on L100

File: integrationtests/testdata/accountsESDTWithTokenType/account-esdt.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
 	"token": "SEMI-abcd",
 	"identifier": "SEMI-abcd-02",
 	"tokenNonce": 2,
-	"properties": "6f6b",
+	"properties": "ok",
```
[the remark this answers is not in the document]
Comment (miiu96) [reply integrationtests/testdata/accountsESDTWithTokenType/account-esdt.json:8]: reverted the changed.

File: process/elasticproc/accounts/accountsProcessor_test.go [hunk 3]
```diff
@@ -497,3 +497,29 @@ func TestAddAdditionalDataIntoAccounts(t *testing.T) {
...
+	invalidHex := "invalid"
+	result = isFrozen(invalidHex)
+	require.False(result)
+
+	// Test with empty input
+	emptyHex := ""
+	result = isFrozen(emptyHex)
+	require.False(result)
+
+	// Test with valid input that returns true
```
[the remark this answers is not in the document]
Comment (miiu96) [reply process/elasticproc/accounts/accountsProcessor_test.go:521]: changed.

File: process/elasticproc/converters/tokenMetaData.go [hunk 4]
```diff
@@ -73,8 +73,16 @@ func PrepareNFTUpdateData(buffSlice *data.BufferSlice, updateNFTData []*data.NFT
 		}
 
 		metaData := []byte(fmt.Sprintf(`{"update":{ "_index":"%s","_id":"%s"}}%s`, index, id, "\n"))
-		base64Attr := base64.StdEncoding.EncodeToString(nftUpdate.NewAttributes)
+		freezeOrUnfreezeTokenIndex := (nftUpdate.Freeze || nftUpdate.UnFreeze) && !accountESDT
```
[the remark this answers is not in the document]
Comment (miiu96) [reply process/elasticproc/converters/tokenMetaData.go:76]: changed

File: .github/workflows/pr-integration-tests.yml [hunk 5]
```diff
@@ -8,7 +8,7 @@ on:
 
 jobs:
   test-1:
-    name: Elasticsearch v7.16.2
+    name: Elasticsearch v7.17.8
```
[the remark this answers is not in the document]
Comment (miiu96) [reply .github/workflows/pr-integration-tests.yml:11]: added

File: integrationtests/logsCrossShard_test.go [hunk 6]
```diff
@@ -34,6 +34,8 @@ func TestIndexLogSourceShardAndAfterDestinationAndAgainSource(t *testing.T) {
 	address1 := "erd1ju8pkvg57cwdmjsjx58jlmnuf4l9yspstrhr9tgsrt98n9edpm2qtlgy99"
 	address2 := "erd1w7jyzuj6cv4ngw8luhlkakatjpmjh3ql95lmxphd3vssc4vpymks6k5th7"
 
+	logID := hex.EncodeToString([]byte("cross-log"))
+
 	// INDEX ON SOURCE
```
[the remark this answers is not in the document]
Comment (miiu96) [reply integrationtests/logsCrossShard_test.go:39]: changed

File: process/elasticproc/accounts/accountsProcessor.go [hunk 7]
```diff
@@ -276,7 +277,7 @@ func (ap *accountsProcessor) getESDTInfo(accountESDT *data.AccountESDT) (*big.In
 
 	tokenMetaData := converters.PrepareTokenMetaData(accountTokenData.MetaData)
 
-	return value, hex.EncodeToString([]byte(accountTokenData.Properties)), tokenMetaData, nil
+	return value, accountTokenData.Properties, tokenMetaData, nil
```
[the remark this answers is not in the document]
Comment (miiu96) [reply process/elasticproc/accounts/accountsProcessor.go:280]: The `encode-to-string` part is done at the node level.

File: process/elasticproc/logsevents/nftsPropertiesProcessor.go (hunk 1 above)
Comment (miiu96) [reply process/elasticproc/logsevents/nftsPropertiesProcessor.go:98]: done

**Review by bogdan-rosianu — APPROVED** [review]

File: integrationtests/accountsESDTRollback_test.go [hunk 8]
```diff
@@ -26,7 +26,7 @@ func TestAccountsESDTDeleteOnRollback(t *testing.T) {
 
 	esdtToken := &esdt.ESDigitalToken{
 		Value:      big.NewInt(1000),
-		Properties: []byte("ok"),
+		Properties: []byte("3032"),
```
Comment (iulianpascalau) [inline integrationtests/accountsESDTRollback_test.go:29]: why not testing with `6f6b` **as the original test was**? 
In fact this is sent by the node as `hex("ok")`

File: process/elasticproc/accounts/accountsProcessor_test.go [hunk 9]
```diff
@@ -175,7 +175,7 @@ func TestGetESDTInfoNFTWithMetaData(t *testing.T) {
 	balance, prop, metaData, err := ap.getESDTInfo(wrapAccount)
 	require.Nil(t, err)
 	require.Equal(t, big.NewInt(1), balance)
-	require.Equal(t, "ok", prop)
+	require.Equal(t, "6f6b", prop)
```
Comment (iulianpascalau) [inline process/elasticproc/accounts/accountsProcessor_test.go:178]: inconsistent behavior: some tests get `3032` other `6f6b`

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**